gulpfile.ts

```diff
@@ -97,20 +121,86 @@ function buildCardObject(filename: string, meta: CardMeta, content: string): Sin
   };
 }
 
-function readdirPromise(path) {
+function readdirPromise(path): Promise<string[]> {
   return new Promise<string[]>((res, rej) => {
-    readdir(path, (err, files) => {
-      if (err) { rej(err); }
+    fs.readdir(path, (err, files) => {
+      if (err) { return rej(err); }
       res(files);
     });
   });
 }
 
 function writeFilePromise(path, content) {
   return new Promise((res, rej) => {
-    writeFile(path, content, err => {
+    fs.writeFile(path, content, err => {
       if (err) { rej(err); }
       res();
     });
   });
 }
+
+// Fail-fast abstractions --------------------------------------------------------------------------
+
+function checkDirectoryShape(contents: string[]): Promise<string[]> {
+  const promises: Array<Promise<string>> = [];
+  for (const dir of contents) {
+    promises.push(new Promise(res => {
+      fs.stat(`./cards/${dir}`, (err, stats) => {
+        if (err) { throw err; }
+        if (!stats.isDirectory()) {
+          throw new Error('No files should be located in the first level of the "cards" directory');
+        }
+        fs.readdir(`./cards/${dir}`, (errr, files) => {
+          if (errr) { throw errr; }
+          if (files.indexOf('card.md') === -1) {
+            throw new Error(`No card.md file found in directory ${dir}`);
+          }
+          res(dir);
+        });
+      });
+    }));
+  }
+  return Promise.all(promises);
+}
+
+function checkCardAttributes(attrs: Partial<CardMeta>, cardName: string): void {
+  const keys = [
+    'authors', 'categories', 'created', 'title', 'updates',
+  ];
+  const attributeKeys = Object.keys(attrs).filter(k => ['pubmed', 'googlescholar'].indexOf(k) === -1);
```

Comment:
I'm fine with removing them. I figured if we were going through cards with a fine-toothed comb it would be useful, but workflow not going how I thought/predicted.